snappi/snappicommon.py

```diff
@@ -159,6 +274,7 @@ def _encode(self):
         """Helper method for serialization
         """
         output = {}
+        self.validate()
```

Comment:
Currently, we're looping over properties multiple times - inside `encode` and `_validate_types`. 

We can loop over properties once and perform encode & _validate_types on per property basis.

Same for `decode`.